waspc/src/Wasp/Generator/Job/Process.hs

```diff
@@ -97,8 +97,10 @@ runNodeCommandAsJob = runNodeCommandAsJobWithExtraEnv []
 runNodeCommandAsJobWithExtraEnv :: [(String, String)] -> Path' Abs (Dir a) -> String -> [String] -> J.JobType -> J.Job
 runNodeCommandAsJobWithExtraEnv extraEnvVars fromDir command args jobType chan =
   NodeVersion.getAndCheckNodeVersion >>= \case
-    Left errorMsg -> exitWithError (ExitFailure 1) (T.pack errorMsg)
-    Right _ -> do
+    NodeVersion.VersionCheckFail errorMsg -> exitWithError (ExitFailure 1) (T.pack errorMsg)
+    -- We don't print the warning message here because `wasp CLI` already likely printed it,
+    -- so it would be too much noise.
```

Comment:
This is a function that executes the node command. So before doing that, it needs to check if node version is ok.
Before we didn't have the check at the Wasp CLI top level, so this was important. Now we have that check, but still, we kept this one here also, to avoid assuming too much and to make this function standalone and independent of what checks the CLI does.
That said, I wanted to avoid printing same warning twice, which is why I left this comment. That is why it says "likely" -> that might change, or it might be called from some other place, in which case this check here is important and maybe that message before is not printed but that is not terrible.

We could refactor the whole system to pass on if node version was checked earlier, but it is not worth it at the moment.